syntax = "proto3";

package platform.agent.v1;

import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";
import "platform/common/v1/types.proto";
import "platform/agent/v1/block.proto";
import "platform/tool/v1/tool_call.proto";

message AgentRun {
  string id = 1;
  string agent_id = 2;
  google.protobuf.Timestamp started_at = 3;
  google.protobuf.Timestamp finished_at = 4;
  repeated Block blocks = 5;
  optional platform.tool.v1.ForceFinalToolCall force_final_tool_call = 6;
  platform.common.v1.Status status = 7;
  repeated string dataset_ids = 8;
  AgentRunMetrics metrics = 9;
  optional string error_message = 10;
}

message AgentRunMetrics {
  int32 total_tokens = 1;
  int32 prompt_tokens = 2;
  int32 completion_tokens = 3;
  google.protobuf.Duration total_duration = 4;
  int32 tool_call_count = 5;
}

